app/frontend/testnet/libs/chainlib-wrapper.js

```diff
@@ -0,0 +1,225 @@
+import * as lib from '@emurgo/js-chain-libs'
+import {bech32_encode, bech32_decode, buf2hex, parse_bech32} from './bech32.js'
+import assert from 'assert'
+import _ from 'lodash'
+
+const {
+  Account,
+  Address,
+  AddressDiscrimination,
+  Fee,
+  Fragment,
+  FragmentId,
+  Hash,
+  Input,
+  InputOutputBuilder,
+  OutputPolicy,
+  Payload,
+  PayloadAuthData,
+  DelegationType,
+  DelegationRatio,
+  PoolDelegationRatios,
+  PoolDelegationRatio,
+  PoolId,
+  Certificate,
+  StakeDelegation,
+  StakeDelegationAuthData,
+  AccountBindingSignature,
+  PrivateKey,
+  PublicKey,
+  SpendingCounter,
+  TransactionBuilder,
+  UtxoPointer,
+  Value,
+  Witness,
+  Witnesses,
+  TransactionSignDataHash,
+} = lib
+
+const hex2buf = (str) => Buffer(str, 'hex')
+
+export const calculate_fee = ({input_count, output_count, cert_count, chain_config}) => {
+  // Fee (#inputs + #outputs) * coefficient + #certificates*certificate + constant
+  const fees = chain_config.fees
+
+  return (
```

Comment:
FYI this is wrong. Jormungandr has a special fee depending on the kind of certificate in the transaction. In Yoroi we have a hack so that it works for StakeDelegation certificates but I will make a proper fix for it this weekend. This is also why in Yoroi -- instead of hard-coding the fee like this -- we calculate the fee by building the transaction in Rust and calling the WASM function to get the fee

Reply:
This is interesting. Especially given the fact that chain settings jormungandr returns mention only the three coefficients.